core/chains/evm/txmgr/evm_tx_store.go

```diff
@@ -1221,6 +1244,75 @@ func (o *evmTxStore) SaveInProgressAttempt(ctx context.Context, attempt *TxAttem
 	return nil
 }
 
+func (o *evmTxStore) GetNonFatalTransactions(ctx context.Context) (txes []*Tx, err error) {
+	var cancel context.CancelFunc
+	ctx, cancel = o.mergeContexts(ctx)
+	defer cancel()
+	qq := o.q.WithOpts(pg.WithParentCtx(ctx))
+	err = qq.Transaction(func(tx pg.Queryer) error {
+		stmt := `
+SELECT * FROM evm.txes
+WHERE state = 'unconfirmed' OR state = 'unstarted' OR state = 'in_progress' 
+   OR state = 'confirmed_missing_receipt' OR state = 'confirmed'
+`
+		var dbEtxs []DbEthTx
+		if err = tx.Select(&dbEtxs, stmt); err != nil {
+			return pkgerrors.Wrap(err, "GetNonFinalizedTransactions failed to load evm.txes")
+		}
+		txes = make([]*Tx, len(dbEtxs))
+		dbEthTxsToEvmEthTxPtrs(dbEtxs, txes)
+		err = o.LoadTxesAttempts(txes, pg.WithParentCtx(ctx), pg.WithQueryer(tx))
+		return pkgerrors.Wrap(err, "GetNonFinalizedTransactions failed to load evm.tx_attempts")
+	}, pg.OptReadOnlyTx())
+
+	return txes, nil
+}
+
+func (o *evmTxStore) GetTxByID(ctx context.Context, id int64) (txe *Tx, err error) {
+	var cancel context.CancelFunc
+	ctx, cancel = o.mergeContexts(ctx)
+	defer cancel()
+	qq := o.q.WithOpts(pg.WithParentCtx(ctx))
+
+	err = qq.Transaction(func(tx pg.Queryer) error {
+		stmt := `SELECT * FROM evm.txes WHERE id = $1`
+		var dbEtxs []DbEthTx
+		if err = tx.Select(&dbEtxs, stmt, id); err != nil {
+			return pkgerrors.Wrap(err, "GetTxByID failed to load evm.txes")
+		}
+		txes := make([]*Tx, len(dbEtxs))
+		dbEthTxsToEvmEthTxPtrs(dbEtxs, txes)
+		if len(txes) != 1 {
+			return fmt.Errorf("GetTxByID failed to get tx with id %v", id)
+		}
+		txe = txes[0]
+		err = o.LoadTxesAttempts(txes, pg.WithParentCtx(ctx), pg.WithQueryer(tx))
+		return pkgerrors.Wrap(err, "GetTxByID failed to load evm.tx_attempts")
+	}, pg.OptReadOnlyTx())
+
+	return txe, nil
+}
+
+func (o *evmTxStore) GetFatalTransactions(ctx context.Context) (txes []*Tx, err error) {
```

Comment:
Is this method only used for testing? If it is, I think we can remove it from the common/tx_store type and just add it to the `TestEvmTxStore` type in the evm_tx_store_test.go file. So we can keep the interfaces in common what's required for functionality